src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.CSharp.cs

```diff
@@ -555,6 +555,103 @@ public R()
 
                 await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, expected: expected, signaturesOnly: signaturesOnly);
             }
+
+            /// <summary>
+            /// This test must be updated when we switch to a new version of the decompiler that supports checked ops.
+            /// </summary>
+            [Theory, CombinatorialData, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+            [WorkItem(42986, "https://github.com/dotnet/roslyn/issues/42986")]
+            public async Task TestCheckedOperators(bool signaturesOnly)
+            {
+                var metadataSource = @"
+public class C
+{
+    public static explicit operator string(C x) => throw new System.Exception();
+
+    public static explicit operator checked string(C x) => throw new System.Exception();
+
+    public static C operator -(C x) => throw new System.Exception();
+
+    public static C operator checked -(C x) => throw new System.Exception();
+
+    public static C operator +(C x, C y) => throw new System.Exception();
+
+    public static C operator checked +(C x, C y) => throw new System.Exception();
+}";
+                var symbolName = "C";
+
+                var expected = signaturesOnly switch
+                {
+                    true => $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public class [|C|]
+{{
+    public C();
+
+    public static C operator +(C x, C y);
+    public static C operator checked +(C x, C y);
+    public static C operator -(C x);
+    public static C operator checked -(C x);
+
+    public static explicit operator string(C x);
+    public static explicit operator checked string(C x);
+}}",
+                    false => $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+// Decompiled with ICSharpCode.Decompiler {ICSharpCodeDecompilerVersion}
+#endregion
+
+using System;
+using System.Runtime.CompilerServices;
+
+public class [|C|]
+{{
+    public static explicit operator string(C x)
+    {{
+        throw new Exception();
+    }}
+
+    [SpecialName]
+    public static string op_CheckedExplicit(C x)
```

Comment:
There is not yet a version of the decompiler that supports checked operators, so for now just asserting the only behavior we have.

Reply:
:+1: That's exactly what I decided to do for static interface members.